engine/classes/Elgg/DataStorageService.php

```diff
@@ -0,0 +1,40 @@
+<?php
+namespace Elgg;
+use Elgg\Filesystem\Adapter\Adapter;
+
+class DataStorageService {
+	private $logger;
+	private $adapter;
+	
+	// @todo file/dir object cache.
+	
+	
+	public function __construct(Logger $logger) {
+		$this->logger = $logger;
+	}
+	
+	public function setAdapter(Adapter $adapter) {
```

Comment:
You can't pass constructor arguments to a service from _elgg_service() because it's already instantiated. Maybe I don't understand what you're suggesting.